src/sacn/private/common.h

```diff
@@ -353,6 +363,84 @@ typedef struct SacnRecvThreadContext
   uint8_t recv_buf[SACN_MTU];
 } SacnRecvThreadContext;
 
+/******************************************************************************
+ * Types used by the sACN Source module
+ *****************************************************************************/
+
+typedef struct SacnSource SacnSource;
+struct SacnSource
+{
+  sacn_source_t handle;  // This must be the first struct member.
+
+  EtcPalUuid cid;
+  char name[SACN_SOURCE_NAME_MAX_LEN];
+
+  bool terminating;  // If in the process of terminating all universes and removing this source.
+
+  EtcPalRbTree universes;
+  size_t num_active_universes;  // Number of universes to include in universe discovery packets.
+  EtcPalTimer universe_discovery_timer;
+  bool process_manually;
+  sacn_ip_support_t ip_supported;
+  int keep_alive_interval;
+  size_t universe_count_max;
+
+  EtcPalRbTree netints;  // Provides a way to look up netints being used by any universe of this source.
```

Comment:
When I was implementing universe discovery, I ran into the issue that netints are specific to each universe, but universe discovery encompasses all universes in a source. The approach I decided to go with was to broadcast universe discovery on the set of network interfaces of all universes on the source. To avoid duplicates I ended up going with a rbtree for this. You can see how this is utilized in the send_universe_discovery function.

Reply:
I improved the documentation for this in https://github.com/ETCLabs/sACN/pull/32/commits/85fd15bb8b82137598220a8bebf033f6fa61fdf5.